From 3f6b2390102ea9d808c7075b5aaeec743b7d104e Mon Sep 17 00:00:00 2001
From: Thomas Dziedzic <gostrc@gmail.com>
Date: Tue, 10 Apr 2012 19:30:35 -0500
Subject: [PATCH 1/2] Backport for mongodb 2.0.4 "Ignore --fork and --logpath
 when using --shutdown SERVER-5186 SERVER-5187"

https://github.com/mongodb/mongo/commit/fa2e2ef95c5c934eef63fff4b98d5eb060c5653923

Backported in preparation for backporting the fix for https://jira.mongodb.org/browse/SERVER-5358
---
 db/cmdline.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/db/cmdline.cpp b/db/cmdline.cpp
index fd759a7..346a9ae 100644
--- a/db/cmdline.cpp
+++ b/db/cmdline.cpp
@@ -239,7 +239,7 @@ namespace mongo {
             cmdLine.noUnixSocket = true;
         }
 
-        if (params.count("fork")) {
+        if (params.count("fork") && !params.count("shutdown")) {
             if ( ! params.count( "logpath" ) ) {
                 cout << "--fork has to be used with --logpath" << endl;
                 ::exit(-1);
@@ -304,7 +304,7 @@ namespace mongo {
         }
 
 #endif
-        if (params.count("logpath")) {
+        if (params.count("logpath") && !params.count("shutdown")) {
             if ( logpath.size() == 0 )
                 logpath = params["logpath"].as<string>();
             uassert( 10033 ,  "logpath has to be non-zero" , logpath.size() );
-- 
1.7.10

